Lib/fontTools/ttLib/tables/_k_e_r_n.py

```diff
@@ -85,80 +89,142 @@ def fromXML(self, name, attrs, content, ttFont):
 		if format not in kern_classes:
 			subtable = KernTable_format_unkown(format)
 		else:
-			subtable = kern_classes[format]()
+			apple = self.version == 1.0
+			subtable = kern_classes[format](apple)
 		self.kernTables.append(subtable)
 		subtable.fromXML(name, attrs, content, ttFont)
 
 
 class KernTable_format_0(object):
 
+	# 'version' is kept for backward compatibility
+	version = format = 0
+
+	def __init__(self, apple=False):
```

Comment:
The apple is really an input to decompile() and compile(), not a property of the table itself.  What do you think about removing it and moving it there? (like isCFF2 if you remember that discussion.)

Also, the way I implemented this in HarfBuzz, I separated the Apple/MS subtable wrapper abstraction from the subtable types themselves.  I think we can do something similar here to simplify the code if you feel like.

Or you merge this, and I do that when implementing format-2.

Here's the HB code BTW:
https://github.com/behdad/harfbuzz/blob/master/src/hb-ot-kern-table.hh

Reply:
> I see Win10 calibri.ttf has 160kb kern table. I suppose MS started ignoring the binary search header completely and use the subtable length instead. We should do the same. I'll fix HarfBuzz as well.

I keep confusing myself.  Even subtable length is being ignored.  The whole table length is used as subtable length.  Argh!

Reply:
I thought about putting "apple" as an argument for decompile/compile, but then I needed that attribute inside fromXML and toXML as well to decide whether to read/write the tupleIndex...
I think I shall leave it like that until we add more formats.